streams/src/test/java/org/apache/kafka/streams/StreamsConfigTest.java

```diff
@@ -264,6 +264,72 @@ public void shouldSupportNonPrefixedProducerConfigs() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        final Map<String, Object> consumerConfigs = streamsConfig.getConsumerConfigs("groupId", "clientId");
+        assertEquals("host", consumerConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldSupportCustomNonPrefixedPropertiesThatAreNotPartOfRestoreConsumerConfig() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        final Map<String, Object> consumerConfigs = streamsConfig.getRestoreConsumerConfigs("clientId");
+        assertEquals("host", consumerConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldSupportCustomNonPrefixedPropertiesThatAreNotPartOfProducerConfig() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        final Map<String, Object> producerConfigs = streamsConfig.getProducerConfigs("clientId");
+        assertEquals("host", producerConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldAddCustomNonPrefixedPropertiesToProducerAndConsumerConfig() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put("custom.property.host", "host");
+        final Map<String, Object> consumerConfigs = streamsConfig.getConsumerConfigs("groupId", "clientId");
+        final Map<String, Object> producerConfigs = streamsConfig.getProducerConfigs("clientId");
+        assertEquals("host", consumerConfigs.get("custom.property.host"));
+        assertEquals("host", producerConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldAddConsumerPrefixedCustomPropertiesToConsumerConfigOnly() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put(consumerPrefix("custom.property.host"), "host");
+        final Map<String, Object> consumerConfigs = streamsConfig.getConsumerConfigs("groupId", "clientId");
+        final Map<String, Object> producerConfigs = streamsConfig.getProducerConfigs("clientId");
+        assertEquals("host", consumerConfigs.get("custom.property.host"));
+        assertNull(producerConfigs.get("custom.property.host"));
+    }
+
+    @Test
+    public void shouldAddProducerPrefixedCustomPropertiesToProducerConfigOnly() {
+        final StreamsConfig streamsConfig = new StreamsConfig(props);
+        props.put(producerPrefix("custom.property.host"), "host");
+        final Map<String, Object> consumerConfigs = streamsConfig.getConsumerConfigs("groupId", "clientId");
+        final Map<String, Object> producerConfigs = streamsConfig.getProducerConfigs("clientId");
+        assertNull(consumerConfigs.get("custom.property.host"));
+        assertEquals("host", producerConfigs.get("custom.property.host"));
```

Comment:
as above -- add check for two missing clients

Reply:
Ok